exasol_data_science_utils_python/model_utils/reservoir_shuffle.py

```diff
@@ -0,0 +1,109 @@
+from typing import Iterator
+
+import numpy as np
+import pandas as pd
+
+
+class ReservoirShuffle:
+    def __init__(self, it: Iterator[pd.DataFrame], buffer_size: int, batch_size: int):
+        self.it = it
+        self.fill_level = 0
+        self.internal_buffer_size = buffer_size + batch_size
```

Comment:
If it's internal why don't you mark it as private:
```suggestion
        self._internal_buffer_size = buffer_size + batch_size
```

Reply:
yes, code style wasn't done yet, will change that. Wanted to get first the logic correct.